lib/diagnostic.js

```diff
@@ -28,6 +46,21 @@ const newArray = function newArray(length, defaultValue) {
 // function then the elements in the array will be set
 // to the result of invoking defaultValue with the
 // elements index as an argument
+const newArray = function newArray(length, defaultValue) {
+  let ary = new Array(length);
+  if (typeof defaultValue === 'function') {
+    for (let i =0; i < length; i++) {
+      ary.forEach(defaultValue[i]);
```

Comment:
You're not using `forEach` correctly. It expects a function as an argument, and executes that function once for each element in the array; additionally, it takes that element as an argument. If you want to use `forEach` in this problem, it would be _instead_ of your `for` loop.

Since you're already using a `for` loop here, all you need to do is write `ary[i] = defaultValue(i)`, much like you wrote `ary[i] = defaultValue` below.